packages/wallet-apis/src/provider.ts

```diff
@@ -0,0 +1,207 @@
+import {
+  toHex,
+  type TypedDataDefinition,
+  type Address,
+  type WalletRpcSchema,
+  type Prettify,
+  type EIP1193RequestFn,
+  type EIP1193Events,
+} from "viem";
+import { AccountNotFoundError, BaseError } from "@alchemy/common";
+import type { ExtractRpcMethod, BaseWalletClient } from "./types.js";
+import { ChainNotFoundError } from "../../../aa-sdk/core/src/errors/client.js";
+import type { PrepareCallsParams } from "./actions/prepareCalls.js";
+import type { SmartWalletActions } from "./decorators/smartWalletActions.js";
+import type { WalletServerViemRpcSchema } from "@alchemy/wallet-api-types/rpc";
+import EventEmitter from "events"; // TODO(v5): do we need to polyfill this for browser?
+
+export type SmartWalletClient1193Methods = [
+  ExtractRpcMethod<WalletRpcSchema, "eth_chainId">,
+  ExtractRpcMethod<WalletRpcSchema, "eth_accounts">,
+  ExtractRpcMethod<WalletRpcSchema, "personal_sign">,
+  ExtractRpcMethod<WalletRpcSchema, "eth_signTypedData_v4">,
+  ExtractRpcMethod<WalletRpcSchema, "eth_sendTransaction">,
+  ExtractRpcMethod<WalletRpcSchema, "wallet_sendCalls">,
+  // We automatically get wallet_getCallsStatus & wallet_getCapabilities
+  // from the underlying rpc w/o any translation.
+  ...WalletServerViemRpcSchema,
+];
+
+export type SmartWalletClientEip1193Provider = Prettify<
+  EIP1193Events & {
+    request: EIP1193RequestFn<SmartWalletClient1193Methods>;
+  }
+>;
+
+export const createEip1193ProviderFromClient = <
+  TAccount extends Address | undefined = Address | undefined,
+>(
+  client: BaseWalletClient<SmartWalletActions<TAccount>>,
+) => {
+  // TODO(v5): implement any other supported events: https://eips.ethereum.org/EIPS/eip-1193#events
+  const eventEmitter = new EventEmitter();
+
+  // TODO(v5): revisit if we actually want to auto-connect the provider like this.
+  (async () => {
```

Comment:
just curious, is auto-connecting the default behavior / part of the spec, or something that we do for convenience?